### ISSUE TITLE
[REF-2391] Support `None` to be passed as a prop

### ISSUE THREAD

**Issue by picklelo** [reflex-dev/reflex#2932]
Currently we cannot pass `None` to a prop. We should support this, and treat it the same as if no argument was passed.

<sub>From [SyncLinear.com](https://synclinear.com) | [REF-2391](https://linear.app/reflex-dev/issue/REF-2391/support-none-to-be-passed-as-a-prop)</sub>

**Comment by dapomeranz** [comment]
`and treat it the same as if no argument was passed`

This is an easy change that I actually almost submitted. Still could. But the harder change is being able pass None to override a non-None default. I have no examples where this is necessary but it could be.

**Comment by picklelo** [comment]
@dapomeranz that's a good call out. I suppose we can start with treating it as no argument and later potentially add support for overriding a default. Though generally if you're overriding, you can set the value you want it to be (since the default should be the default).

**Comment by AmanSal1** [comment]
@picklelo Could this be a good issue for a first-time contributor to work on, or do you have any other suggestions ?

**Comment by picklelo** [comment]
@AmanSal1 yes I think this would be a good issue! I'm hoping it just is a matter of checking checking the `props` dictionary in the `Component.create` method here: https://github.com/reflex-dev/reflex/blob/e2eced79b1cc5de95e30f29b9573d6fde2d801cf/reflex/components/component.py#L604

and filtering out all the None props before calling `cls(children=children, **props)` at the end. We can add a unit test in `test_component.py` where we try to pass in `None` props and make sure it just filters them out.

**Comment by AmanSal1** [comment]
@picklelo  So, in the component.py file, I suppose there's a one-line change needed. It requires iterating over the prop dictionary and filtering out any None values. However, I'm a bit confused about the associated test case. Can you tell if this is the right test case?

```
def test_create_component_with_none_props(component1):
    """Test that the component is created correctly with None props filtered out.

    Args:
        component1: A test component.
    """
    children = [component1() for _ in range(3)]
    attrs = {"color": "white", "text_align": "center", "font_size": None}
    c = component1.create(*children, **attrs)
    assert isinstance(c, component1)
    assert c.children == children
    assert c.style == {"color": "white", "textAlign": "center"}
    assert "font_size" not in c.props

```

**Comment by picklelo** [comment]
@AmanSal1 yes that test case looks correct to me. Before your change, I would expect it to fail with some error related to `None` not being a valid type for the prop.

One change though - in your `TestComponent` you should specify the props to match your dictionary in your test.

```python
class TestComponent(Component):
        prop1: str
        prop2: str
        ...
```

Any prop not specifically listed in the Component will be a style (CSS) prop. Let's add one of those in the test case too.

**Comment by AmanSal1** [comment]
> I would expect it to fail with some error related to None not being a valid type for the prop. 

@picklelo Yes, it is failing with a type error. Additionally, I believe the component is returning an empty set because when I check for the style tag, it appears to be empty.

> FAILED test_component.py::test_invalid_prop_type[None-1] - Failed: DID NOT RAISE <class 'TypeError'>
FAILED test_component.py::test_create_filters_none_props - AssertionError: assert 'style' in set()

```
@pytest.fixture
def test_component():
    class TestComponent(Component):
        pass

    return TestComponent


# Write a test case to check if the create method filters out None props
def test_create_filters_none_props(test_component):
    child1 = test_component()
    child2 = test_component()
    props = {
        'prop1': 'value1',
        'prop2': None,
        'prop3': 'value3',
        'prop4': None,
        'style': {'color': 'red', 'font-size': '12px'}  # Adding a style prop
    }

    component = test_component.create(child1, child2, **props)

    # Assert that None props are not present in the component's props
    assert 'prop2' not in component.get_props()
    assert 'prop4' not in component.get_props()

    # Assert that the style prop is present in the component's props
    assert 'style' in set(component.get_props())
```

**Comment by AmanSal1** [comment]
@picklelo I have added a pull request can you review.

PR Url - https://github.com/reflex-dev/reflex/pull/2969

**Comment by AmanSal1** [comment]
@picklelo We can close this.

**Comment by picklelo** [comment]
Merged in #2969
